dds/idl/marshal_generator.cpp

```diff
@@ -3129,6 +3220,18 @@ bool marshal_generator::gen_union(AST_Union* node, UTL_ScopedName* name,
     serialized_size.endArgs();
 
     if (has_key) {
+      if (may_be_delimited) {
+        be_global->impl_ <<
+          "  serialized_size_delimiter(encoding, size);\n";
+      }
+
+      if (may_be_parameter_list) {
+        be_global->impl_ <<
+          "  size_t xcdr1_pl_running_total = 0;\n";
+        be_global->impl_ <<
+          "  serialized_size_parameter_id(encoding, size, xcdr1_pl_running_total);\n";
```

Comment:
```suggestion
        be_global->impl_ <<
          "  size_t mutable_running_total = 0;\n";
          "  serialized_size_parameter_id(encoding, size, mutable_running_total);\n";
```

Reply:
And you will also need a call to `serialized_size_list_end_parameter_id` at the end of this function too.